apps/frontend/core-dapp/src/features/trade/close-trade/CloseTradeStore.ts

```diff
@@ -278,33 +280,69 @@ export class CloseTradeStore {
 
   // fees and rewards
 
-  private get fees(): FeeStore[] | undefined {
+  private get uniswapFeeStore(): FeeStore | undefined {
     const { selectedMarket } = this
     if (selectedMarket === undefined) return undefined
 
-    if (selectedMarket.resolved) {
-      const { redemptionFee, tokenSender } = selectedMarket
-      if (redemptionFee === undefined || tokenSender === undefined) return undefined
-
-      return [
-        new FeeStore({
-          applicableToWstEth: () => this.amountOutInWstEthBN,
-          fee: redemptionFee,
-          tokenSender,
-        }),
-      ]
+    const fee = selectedMarket.resolved
+      ? fraction(BigNumber.from(0), BigNumber.from(1))
+      : TradeStore.UNISWAP_FEE
+
+    const reverseFee = (amountAfterFee: BigNumber | undefined): BigNumber | undefined => {
+      if (amountAfterFee === undefined) return undefined
+
+      return div(
+        amountAfterFee,
+        sub(fraction(BigNumber.from(1), BigNumber.from(1)), TradeStore.UNISWAP_FEE)
+      )
     }
 
+    return new FeeStore({
+      // When quoting, Uniswap gives us the amount after fee. To calculate the
+      // correct fee we have to reverse it and then re-apply it.
+      applicableToWstEth: () => reverseFee(this.amountOutInWstEthBN),
+      fee,
+      tokenSender: undefined,
+    })
```

Comment:
I think writing the function in applicableToWstEth is more readable and cleaner since we don't have to refer back to `reverseFee` to check again the applicableToWstEth usage? wdyt?

```suggestion
    const fee = selectedMarket.resolved
      ? fraction(BigNumber.from(0), BigNumber.from(1))
      : TradeStore.UNISWAP_FEE

    return new FeeStore({
      applicableToWstEth: () => {
        if (this.amountOutInWstEthBN === undefined) return undefined

        // Get amount before fee from output of Uniswap's quote, which is after fee.
        return div(
          this.amountOutInWstEthBN,
          sub(fraction(BigNumber.from(1), BigNumber.from(1)), TradeStore.UNISWAP_FEE)
        )
      },
      fee,
      tokenSender: undefined,
    })
```

Reply:
Sure, I actually did it this way first but then switched back because I thought it'd improve readability haha but I have no strong opinion

Reply:
thanks!